Source/Plugins/EditorBase/PropertyEditors/ObjectRefPropertyEditor.cs

```diff
@@ -322,26 +329,61 @@ protected override void OnMouseDown(MouseEventArgs e)
 		protected override void OnMouseUp(MouseEventArgs e)
 		{
 			base.OnMouseUp(e);
+
+			bool leftMouseButtonUp = (e.Button & MouseButtons.Left) != MouseButtons.None;
+
 			this.panelDragBegin = Point.Empty;
-			if (this.buttonResetPressed && (e.Button & MouseButtons.Left) != MouseButtons.None)
+			if (this.buttonResetPressed && leftMouseButtonUp)
 			{
 				if (this.buttonResetPressed && this.buttonResetHovered) this.ResetReference();
 				this.buttonResetPressed = false;
-				this.Invalidate();
 			}
-			else if (this.buttonShowPressed && (e.Button & MouseButtons.Left) != MouseButtons.None)
+			else if (this.buttonShowPressed && leftMouseButtonUp)
 			{
-				if (this.buttonShowPressed && this.buttonShowHovered) this.ShowReferencedContent();
+				if (this.buttonShowPressed && this.buttonShowHovered)
+				{
+					ShowContentSelectionDialog();
+				}
+
 				this.buttonShowPressed = false;
-				this.Invalidate();
 			}
+			else if (this.panelHovered && leftMouseButtonUp)
+			{
+				this.ShowReferencedContent();
+			}
+
+			this.Invalidate();
 		}
 		protected override void OnMouseDoubleClick(MouseEventArgs e)
 		{
 			base.OnMouseDoubleClick(e);
+
 			if (this.panelHovered && (e.Button & MouseButtons.Left) != MouseButtons.None)
-				this.ShowReferencedContent();
+			{
+				ShowContentSelectionDialog();
+			}
 		}
+
+		protected void ShowContentSelectionDialog()
+		{
+			var tmpResourceSelectionForm = new ContentSelectionDialog
+			{
+				FilteredType = this.ReferenceType,
+				ResourcePath = this.ReferenceName
+			};
+
+			var result = tmpResourceSelectionForm.ShowDialog();
+
+			if (result == DialogResult.OK)
+			{
+				var tmpDataObject = new DataObject();
+
+				tmpResourceSelectionForm.SerializeToData(tmpDataObject);
```

Comment:
I was hoping to keep this as minimal a change as possible, and given the way paste, copy, drag and drop are used I'm leery of deviating too far. Can you elaborate further on what you would like to see?

Reply:
Sure! So the reason for serialization at the point where I assume you copied it from is that we're stashing things in the Windows clipboard via Ctrl+C / Ctrl+V. Since this means that data can, in some situations, travel outside of Duality, serialization is a requirement to make this work.

With the new dialog, however, there is no need for data to ever leave Duality. Just hand it over like you would normally, e.g. ask the dialog what the user selected directly. One way to go about this would be to add a `SelectedResult` property to the dialog class, which is retrieved where you also check for the `DialogResult`.